test/Log/LevelFilterTest.php

```diff
@@ -0,0 +1,127 @@
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * @category    Test
+ * @package     OpusTest\Log
+ * @author      Kaustabh Barman <barman@zib.de>
+ * @copyright   Copyright (c) 2020, OPUS 4 development team
+ * @license     http://www.gnu.org/licenses/gpl.html General Public License
+ */
+
+namespace OpusTest\Log;
+
+use Opus\Log\LevelFilter;
+
+class LevelFilterTest extends \PHPUnit_Framework_TestCase
+{
+    public function testSetLevel()
+    {
+        $filter = new LevelFilter(\Zend_Log::WARN);
+        $filter->setLevel(\Zend_Log::INFO);
+
+        $infoEvent = ['priority' => \Zend_Log::INFO];
+        $debugEvent = ['priority' => \Zend_Log::DEBUG];
+
+        $this->assertTrue($filter->accept($infoEvent));
+        $this->assertFalse($filter->accept($debugEvent));
+    }
+
+    public function testSetLevelOnNullArgument()
+    {
+        $filter = new LevelFilter(\Zend_Log::WARN);
+        $filter->setLevel(null);
+
+        $emergEvent = ['priority' => \Zend_Log::EMERG];
+        $debugEvent = ['priority' => \Zend_Log::DEBUG];
+
+        $this->assertTrue($filter->accept($emergEvent));
+        $this->assertTrue($filter->accept($debugEvent));
+    }
+
+    public function testSetLevelChangedAfterNullArgument()
+    {
+        $filter = new LevelFilter(\Zend_Log::WARN);
+        $filter->setLevel(null);
+        $filter->setLevel(\Zend_Log::INFO);
+
+        $this->assertTrue($filter->accept(['priority' => \Zend_Log::INFO]));
+        $this->assertSame(\Zend_Log::INFO, $filter->getLevel());
+    }
+
+    public function testSetLevelOnNegativeArgument()
+    {
+        $filter = new LevelFilter(\Zend_Log::WARN);
+
+        $exceptionMessage = 'Level should be of Integer type and cannot be negative';
+
+        $this->setExpectedException(\InvalidArgumentException::class, $exceptionMessage);
+
+        $filter->setLevel(-1);
+    }
+
+    public function testSetLevelOnStringArgument()
+    {
+        $filter = new LevelFilter(\Zend_Log::WARN);
+
+        $exceptionMessage = 'Level should be of Integer type and cannot be negative';
+
+        $this->setExpectedException(\InvalidArgumentException::class, $exceptionMessage);
+
+        $filter->setLevel('TestLevel');
+    }
+
+    public function testSetLevelStringNumericArgument()
+    {
+        $filter = new LevelFilter(\Zend_Log::WARN);
+
+        $filter->setLevel('7');
+
+        $this->assertTrue($filter->accept(['priority' => \Zend_Log::DEBUG]));
+        $this->assertEquals(\Zend_Log::DEBUG, $filter->getLevel());
+    }
+
+    public function testGetLevel()
+    {
+        $filter = new LevelFilter(\Zend_Log::WARN);
+
+        $this->assertEquals(\Zend_Log::WARN, $filter->getLevel());
+    }
+
+    public function testGetLevelOnNull()
+    {
+        $filter = new LevelFilter(\Zend_Log::WARN);
+        $filter->setLevel(null);
+
+        $this->assertNull($filter->getLevel());
+    }
+
+    public function testGetLevelOnInvertedConfig()
```

Comment:
I don't get the name, "InvertedConfig"? I know why this test exists, but imagine a new developer trying to understand it. Is it still necessary? It is kind of testing for the absence of an implementation we rejected. It is a bit like `testGetLevel` just with a different level and an operator. I mean okay, theoretically we could test for every level if it is returned as well. Where is the limit of what makes sense?